src/Microsoft.OData.Core/Json/JsonValueUtilsAsync.cs

```diff
@@ -468,8 +464,10 @@ internal static async Task WriteBinaryStringAsync(this TextWriter writer, byte[]
             // write remaining bytes in buffer
             if (bufferIndex > 0)
             {
-                await writer.WriteAsync(buffer.Value, 0, bufferIndex).ConfigureAwait(false);
+	            return writer.WriteAsync(buffer.Value, 0, bufferIndex);

```

Comment:
@henning-krause Eliding async/await might not advisable in this method since both `RentFromBuffer` method (called by `InitializeBufferIfRequired`) and `WriteEscapedCharArrayToBuffer` can throw exceptions and there would be no state machine to capture them and place them on the returned task. They'll instead be raised directly. The expected asynchronous semantics are that exceptions are placed on the returned task.
However, when the chances of the that exception are extremely rare, we could do a trade off and go for the optimization...

Reply:
I've changed this to catch any exceptions and return a faulted task in this case using
```cs
return TaskUtils.GetFaultedTask(ex);
```
Would that be acceptable?

Reply:
@henning-krause That should be okay. Hopefully the performance penalty for the use of the try...catch is much lower than the penalty for the task allocation

Reply:
@henning-krause Maybe catch `ODataException` to fix the build - `RentFromBuffer` throws an `ODataException`.

Reply:
https://blog.stephencleary.com/2016/12/eliding-async-await.html#:~:text=One%20of%20the%20most%20common%20mistakes%20in%20eliding,example%2C%20eliding%20the%20keywords%20will%20abort%20the%20download.

Reply:
+1 on erring on the side of caution when it comes to eliding non-trivial implementations

Reply:
I've reverted my change.